timemachine/cpp/src/context.cu

```diff
@@ -55,6 +55,12 @@ double Context::_get_temperature() {
     }
 }
 
+void Context::initialize_local_md() {
+    if (!this->local_md_pots_) {
```

Comment:
nit: I prefer to be a bit more verbose in these types of situations, and not rely on operator overloads of `!`

```
if(this->local_md_pots_.get() == nullptr) {}
```